lib/StripeResource.js

```diff
@@ -192,14 +197,50 @@ StripeResource.prototype = {
       callback.call(
         self,
         new Error.StripeConnectionError({
-          message: 'An error occurred with our connection to Stripe',
+          message: self._generateConnectionErrorMessage(requestRetries),
           detail: error,
         }),
         null
       );
     }
   },
 
+  _shouldRetry: function(res, numRetries) {
+    // Retry on connection error, if we have retries left.
+    if (!res) {
+      return numRetries < this._stripe.getMaxNetworkRetries();
+    }
+
+    // Retry on conflict or rate limit error, if we have retries left.
+    if (res.statusCode === 409 || res.statusCode === 429) {
+      return numRetries < this._stripe.getMaxNetworkRetries();
+    }
+
+    return false;
+  },
+
+  _getSleepTimeInMS: function(numRetries) {
```

Comment:
FWIW in JS "time" always means ms to me, so I don't think is necessary, but fine to leave.

Reply:
I like the idea of being explicit here, might help prevent a bug in the future.